app/src/main/java/de/tum/in/tumcampusapp/activities/FeedbackActivity.java

```diff
@@ -0,0 +1,363 @@
+        customEmailView = findViewById(R.id.feedback_custom_email);
+        includeEmail = findViewById(R.id.feedback_include_email);
+        includeLocation = findViewById(R.id.feedback_include_location);
+
+        lrzId = Utils.getSetting(this, Const.LRZ_ID, "");
+
+        initTopicSpinner(savedInstanceState);
+        initIncludeLocation(savedInstanceState);
+        initIncludeEmail(savedInstanceState);
+
+        if(savedInstanceState == null){
+            picPaths = new ArrayList<>();
+        } else {
+            picPaths = savedInstanceState.getStringArrayList(Const.FEEDBACK_PIC_PATHS);
+            feedbackView.setText(savedInstanceState.getString(Const.FEEDBACK_MESSAGE));
+        }
+
+        RecyclerView pictureList = findViewById(R.id.feedback_image_list);
+        pictureList.setLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.HORIZONTAL, true));
+        thumbAdapter = new FeedbackThumbAdapter(picPaths);
+        pictureList.setAdapter(thumbAdapter);
+
+        pattern = Pattern.compile(EMAIL_PATTERN);
+    }
+
+    private void initIncludeLocation(Bundle savedInstanceState){
+        includeLocation.setOnClickListener(view -> {
+            if(includeLocation.isChecked() && location == null){
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                    int permissionCheck = ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_CALENDAR);
+                    if( permissionCheck == PackageManager.PERMISSION_DENIED){
+                        requestPermissions(new String[]{Manifest.permission.ACCESS_FINE_LOCATION}, PERMISSION_LOCATION);
+                    }
+                }
+                saveLocation();
+            }
+        });
+
+        if(savedInstanceState != null){
+            includeLocation.setChecked(savedInstanceState.getBoolean(Const.FEEDBACK_INCL_LOCATION));
+            saveLocation();
+        } else {
+            includeLocation.setChecked(true);
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.ICE_CREAM_SANDWICH_MR1) {
+                includeLocation.callOnClick();
+            } else {
+                saveLocation();
+            }
+        }
+    }
+
+    private void saveLocation(){
+        location = new LocationManager(this).getLastLocation();
+        if(location == null){
+            includeLocation.setChecked(false);
+        }
+    }
+
+    private void initIncludeEmail(Bundle savedInstanceState){
+        if(!"".equals(lrzId)){
+            email = lrzId + "@mytum.de";
+        } else {
+            if(savedInstanceState != null){
+                email = savedInstanceState.getString(Const.FEEDBACK_EMAIL, null);
+            }
+        }
+
+        if(email != null){
+            includeEmail.setText(getResources().getString(R.string.feedback_include_email_tum_id, email));
+            includeEmail.setChecked(true);
+        } else {
+            includeEmail.setChecked(false);
+        }
```

Comment:
I think some of that `email != null` logic can be collapsed with the previous else branch...